tests/serialization/test_dag_serialization.py

```diff
@@ -1654,6 +1673,28 @@ def test_mapped_operator_xcomarg_serde():
     assert xcom_arg.operator is serialized_dag.task_dict['op1']
 
 
+def test_task_resources_serde():
+    """
+    Test task resources serialization/deserialization.
+    """
+    from airflow.operators.dummy import DummyOperator
+
+    execution_date = datetime(2020, 1, 1)
+    task_id = 'task1'
+    with DAG("test_task_resources", start_date=execution_date) as _:
+        task = DummyOperator(task_id=task_id, resources={"cpus": 0.1, "ram": 2048})
+
+    serialized = SerializedBaseOperator._serialize(task)
+    assert serialized['resources'] == (
+        '{'
+        '"cpus": {"name": "CPU", "qty": 0.1, "units_str": "core(s)"}, '
+        '"disk": {"name": "Disk", "qty": 512, "units_str": "MB"}, '
+        '"gpus": {"name": "GPU", "qty": 0, "units_str": "gpu(s)"}, '
+        '"ram": {"name": "RAM", "qty": 2048, "units_str": "MB"}'
+        '}'
+    )
```

Comment:
This shouldn't be a string of a JSON object inside a json object -- it should be the JSON object directly

Reply:
this depends on the protocol, https://github.com/apache/airflow/pull/21445/files#diff-807ca0a4fd53aeb41166621c9842b0f89b7931fc64e9a60befa36c776db45efaR322

we can also use `to_dict` when serializing `resources`.

Reply:
with more thoughts, it is better to use to_dict. changed.